exporter/src/app/renderer/svg.cljs

```diff
@@ -293,88 +294,74 @@
-                     (assoc :query (u/map->query-string params)))]
-
-    (bw/exec!
-     #js {:screen #js {:width bw/default-viewport-width
-                       :height bw/default-viewport-height}
-          :viewport #js {:width bw/default-viewport-width
-                         :height bw/default-viewport-height}
-          :locale "en-US"
-          :storageState #js {:cookies (bw/create-cookies uri {:token token})}
-          :deviceScaleFactor scale
-          :userAgent bw/default-user-agent}
-     (fn [page]
-       (l/info :uri uri)
-       (p/do!
-        (bw/nav! page uri)
-        (p/let [dom (bw/select page "#screenshot")]
-          (bw/wait-for dom)
-          (bw/sleep page 2000))
-
-        (extract page)))))))
-
-(s/def ::name ::us/string)
-(s/def ::suffix ::us/string)
-(s/def ::type #{:svg})
-(s/def ::page-id ::us/uuid)
-(s/def ::file-id ::us/uuid)
-(s/def ::object-id ::us/uuid)
-(s/def ::scale ::us/number)
-(s/def ::token ::us/string)
-(s/def ::uri ::us/uri)
-
-(s/def ::params
-  (s/keys :req-un [::name ::suffix ::type ::object-id ::page-id ::file-id ::scale ::token]
-          :opt-un [::uri]))
-
-(defn render
-  [params]
-  (us/assert ::params params)
-  (p/let [content (render-object params)]
-    {:data content
-     :name (str (:name params)
-                (:suffix params "")
-                ".svg")
-     :size (alength content)
-     :mtype "image/svg+xml"}))
+            (-> (bw/select-all page (str/concat "#screenshot-" id " foreignObject"))
+                (p/then (fn [nodes] (p/all (map (partial extract-txt-node page) nodes))))
+                (p/then (fn [nodes] (d/index-by :id nodes)))))
+
+          (extract-svg [page {:keys [id] :as object}]
+            (let [node (bw/select page (str/concat "#screenshot-" id))]
+              (bw/wait-for node)
+              (bw/eval! node (fn [elem] (.-outerHTML ^js elem)))))
+
+          (prepare-options [uri]
+            #js {:screen #js {:width bw/default-viewport-width
+                              :height bw/default-viewport-height}
+                 :viewport #js {:width bw/default-viewport-width
+                                :height bw/default-viewport-height}
+                 :locale "en-US"
+                 :storageState #js {:cookies (bw/create-cookies uri {:token token})}
+                 :deviceScaleFactor scale
+                 :userAgent bw/default-user-agent})
+
+          (render-object [page {:keys [id] :as object}]
+            (p/let [tmpdir (sh/mktmpdir! "svg-render")
+                    path   (path/join tmpdir (str/concat id (mime/get-extension type)))
+                    node   (bw/select page (str/concat "#screenshot-" id))]
+              (bw/wait-for node)
+              (p/let [xmldata (extract-svg page object)
+                      txtdata (extract-txt-nodes page object)
+                      result  (replace-text-nodes xmldata txtdata)
+                      result  (str/replace result "&nbsp;" "&#160;")]
```

Comment:
Seems that you have removed the comment explaining this (I think it's still necessary)

Reply:
agree